src/daemons/GraphDaemon.cpp

```diff
@@ -110,10 +110,24 @@ int main(int argc, char *argv[]) {
         return EXIT_FAILURE;
     }
 
+    if (FLAGS_num_netio_threads == 0) {
+        FLAGS_num_netio_threads = std::thread::hardware_concurrency();
+    }
     if (FLAGS_num_netio_threads <= 0) {
         LOG(WARNING) << "Number of networking IO threads should be greater than zero";
         return EXIT_FAILURE;
     }
```

Comment:
combine these two conditions ?

My thought: FLAGS_num_netio_threads <= 0 means all cores

Reply:
`0` is to be a legal option and the only option to indicate _number of cores_, while a negative value is not. The second check is to cover the negative AND the failure case(`hard_concurrency` has chances to fail)